### PR TITLE
Removed unused configurations from CodeTools solution.

### PR DESCRIPTION
**Pull request by hubuk**
This is the first part of the solutions/projects configuration rework across entire repository. The main reason for this is to simplify development, maintenance and to make new developer's first impression better. I am splitting entire work into multiple PR's to make review simpler or even possible.

Currently we have following configurations defined for variety of projects:
Debug, Release, Debug64, Devlab64, Devlab9, Devlab9ro, Internal64, Internal9, ClousotDebug, DebugCommandLine, DebugNoData, FullDebug, NoExtendedRuntime, msr9, devlab9ts, devlab9std, msft9.
Most of them are historical remnants of the times where CodeContracts were distributed in two or even three flavors. The aim is to limit them to two: Debug and Release.

Proof of Concept is already available on a separate branch: https://github.com/hubuk/CodeContracts/tree/reconfiguration-poc

### CODE REVIEW

**Comment by sharwell** [comment]
:memo: In CodeToolsUpdate10.csproj configuration Release|AnyCPU, the platform target is actually set to x86.

:memo: In the Release|AnyCPU configuration of both Microsoft.VisualStudio.CodeTools.IPropertyPane10.csproj, Microsoft.VisualStudio.CodeTools.ITaskManager10.csproj, PropertyPage10.csproj, and TaskManager10.csproj, the `DebugType` is set to `full` (normally this is `pdbonly` for release configurations). In the same configurations, optimization is not enabled (it normally is for release configurations).

:memo: In the Release|Win32 configuration of PropertyPageUI10.vcxproj and TaskManagerUi10.vcxproj, several paths refer to the directory **.\Debug**. This matches the behavior of the previous Devlab9|Win32 configuration, but it seems like these should be referencing **.\Release** instead.

:question: Why are we editing the following files instead of simply deleting them?

* CodeToolsUpdate9.csproj
* Microsoft.VisualStudio.CodeTools.IPropertyPane9.csproj
* Microsoft.VisualStudio.CodeTools.ITaskManager9.csproj
* PropertyPage9.csproj
* PropertyPageUI9.vcproj
* TaskManager9.csproj
* TaskManagerUi9.vcproj

**Comment by hubuk** [comment]
I would love to remove old *9.csproj files. If we are OK with that I will include appropriate changes in PRs.

**Comment by sharwell** [comment]
> I would love to remove old *9.csproj files. If we are OK with that I will include appropriate changes in PRs.

To me it just seems like more unused configurations from the same projects. I think it makes sense to delete them as we clean up the projects (e.g. delete CodeToolsUpdate9.csproj when we clean up CodeToolsUpdate10.csproj).

**Comment by sharwell** [comment]
I definitely want to see this pass the AppVeyor build before it gets merged. I also want to see each of the items I mentioned previously considered to make sure the project is configured as we intend. Otherwise this looks nearly complete.

**Comment by SergeyTeplyakov** [comment]
> I definitely want to see this pass the AppVeyor build before it gets merged (I can push a branch to my fork whenever you are ready to see it tested).

Sounds good for me.

**Comment by hubuk** [comment]
@sharwell I have included all the changes you requested. Unfortunately I have rebased my previous changes too early so new fixes are a little bit harder to find. I will wait with rebase next time.

AppVeyor build has passed: https://ci.appveyor.com/project/hubuk/codecontracts/build/1.9.11020.1

There is only one weak point introduced in 3a95da76839f97c662d8d59a1d6fcd95248b97de but it will be fixed later when I submit changes in main solution. Currently it has nearly no impact as the buildMSI10.xml script is not being used with configuration other than devlab9.

**Comment by SergeyTeplyakov** [comment]
@hubuk Looks good for me... @sharwell are you ok to merge?

**Comment by hubuk** [comment]
I am preparing one more commit. After that I will prepare final appveyor build to verify changes.

**Comment by sharwell** [comment]
I'm a bit concerned about copying binaries from the tools build. The main build would still work even if things aren't copied correctly because an old copy of the tools build [remains in source control](https://github.com/Microsoft/CodeContracts/tree/master/Microsoft.Research/ImportedCodeTools). I would be more comfortable if these files were remove from source control, even if it means users need to build one solution (the code tools solution) before they can build the main CodeContracts solution from inside the IDE.

What do you think @hubuk? Note that the files in source control are stale, and if they are used for the final build some old bugs would reappear.

File: CodeContracts.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2013
-VisualStudioVersion = 12.0.31101.0
+# Visual Studio 14
```
Comment (sharwell) [inline CodeContracts.sln]: :exclamation: We are not yet ready to make this change. The last time I checked, CodeContracts.sln has build failures if you try to compile it inside Visual Studio 2015. It should open by default in 2013 until we make a separate decision to move to 2015 for primary development.

Comment (hubuk) [inline CodeContracts.sln]: You are right. I was trying to leave sln file headers unchanged but failed this time. Good catch.
There are some problems with VS2015 build which I am going to address in later time.

**Comment by hubuk** [comment]
@sharwell CodeTools looks like a separate project which was used as a starting point to Visual Studio integration part of Code Contracts. It has its own SDK, installer and samples but they are not useful for CodeContracts as a product. 
Currently there is a VisualStudio Integration folder in CodeContracts solution which contains ManagedContractsVSPropertyPane project. The same project is also part of CodeTools solution. 
This leads me to an idea to move entire content of CodeCools to main solution. Incorporate its WiX parts and remove ImportedCodeTools directory. This will greatly simplify build process and code discoverability also addressing the problem with copying binaries you have mentioned.

**Comment by hubuk** [comment]
Build has passed: https://ci.appveyor.com/project/hubuk/codecontracts/build/1.9.11021.5
@SergeyTeplyakov @sharwell Are you ok with the changes? Should I squash the commits before merge?

**Comment by sharwell** [comment]
I just ran a local build from this branch. During the build, the CodeTools.msm and CodeToolsUpdate.exe files in the ImportedCodeTools directory are updated as expected. However, the IPropertyPane.dll and ITaskManager.dll files in the same directory are not changed. This behavior matches the behavior from before this pull request, so I think this can be merged.

**Comment by hubuk** [comment]
OK, this can be merged as is. Without squashing. I have already started work on next part of the configuration related changes.

**Pull request merged**